src/kava.js

```diff
@@ -273,9 +279,39 @@ class Kava extends BasePlugin {
     }
   }
 
+  /**
+   * calculates the available buffer length
+   * @returns {number} the remaining buffer length of the current played time
+   * @private
+   */
+  _getAvailableBuffer(): number {
+    let retVal = 0;
+    if (this.player.buffered) {
+      const buffered = this.player.buffered;
+      for (let i = 0; i < buffered.length; i++) {
+        // find the relevant buffer time range containing the current time
+        if (buffered.start(i) <= this.player.currentTime && this.player.currentTime <= buffered.end(i)) {
+          retVal = buffered.end(i) - this.player.currentTime;
+        }
+      }
+    }
+    return retVal;
+  }
+
   _onReport(): void {
     if (this._viewEventEnabled) {
       this._updatePlayTimeSumModel();
+      this._model.updateModel({
+        bandwidth:
+          this._totalSegmentsDownloadTime > 0 ? Math.round((this._totalSegmentsDownloadBytes * 8) / this._totalSegmentsDownloadTime) / 1000 : 0,
+        manifestDownloadTime: this._maxManifestDownloadTime,
+        soundMode: this.player.muted ? SoundMode.SOUND_OFF : SoundMode.SOUND_ON,
+        tabMode: document.hasFocus() ? TabMode.TAB_FOCUSED : TabMode.TAB_NOT_FOCUSED,
+        availableBuffer: this._getAvailableBuffer()
+      });
+      this._totalSegmentsDownloadTime = 0;
```

Comment:
check if we should reset this on the model level, accumulate the stats where you get them, and reset them after report